### ISSUE TITLE
calling api's with same tenant using `EnableTokenAcquisitionToCallDownstreamApi` since 2.17.2 result in `MsalClientException`

### ISSUE THREAD

**Issue by EelcoLos** [AzureAD/microsoft-identity-web#2784]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

2.17.2

### Web app

Sign-in users and call web APIs

### Web API

Protected web APIs call downstream web APIs

### Token cache serialization

Distributed caches

### Description

as I'm using the downstream api to call items on entra via an organization api (`/api/me/users`), the error i'm getting is since `2.17.2` is
> {"statusCode":400,"message":"One or more errors occurred!","errors":{"generalErrors":["A authority of type Generic was used at the application and of type B2C at the request level. Please use the same authority type between the two. "]}}

It tries to retrieve the token via `Microsoft.Identity.Web.ITokenAcquisition`
```cs
	private async Task AddHeaders()
	{
		var token = await _tokenAcquisition.GetAccessTokenForUserAsync(
			_config["scopes"]!.Split(' '),
			authenticationScheme: Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectDefaults.AuthenticationScheme);

		_httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);

		_httpClient.DefaultRequestHeaders.Add("Accept", "application/json");
	}
```
but since 2.17.2, this part results into the error

### Reproduction steps

1. have a custom api to retrieve users
2. use an api with `DownstreamApi` to call that api
3. try and collect a token to add the bearer header
4. the error occurs with version `2.17.2` or higher (still with `2.17.5` as of this time)

### Error message

Exception thrown: 'Microsoft.Identity.Client.MsalClientException' in System.Private.CoreLib.dll: 'A authority of type Generic was used at the application and of type B2C at the request level. Please use the same authority type between the two. '
   at Microsoft.Identity.Client.AuthorityInfo.AuthorityInfoHelper.ValidateTypeMismatch(AuthorityInfo configAuthorityInfo, AuthorityInfo requestAuthorityInfo)
   at Microsoft.Identity.Client.AuthorityInfo.AuthorityInfoHelper.<CreateAuthorityForRequestAsync>d__1.MoveNext()
   at Microsoft.Identity.Client.ApplicationBase.<CreateRequestParametersAsync>d__5.MoveNext()
   at Microsoft.Identity.Client.ConfidentialClientApplication.<CreateRequestParametersAsync>d__18.MoveNext()
   at Microsoft.Identity.Client.ApiConfig.Executors.ClientApplicationBaseExecutor.<ExecuteAsync>d__2.MoveNext()
   at Microsoft.Identity.Web.TokenAcquisition.<GetAuthenticationResultForWebAppWithAccountFromCacheAsync>d__28.MoveNext()
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.Identity.Web.TokenAcquisition.<GetAuthenticationResultForUserAsync>d__15.MoveNext()
   at Microsoft.Identity.Web.TokenAcquisition.<GetAccessTokenForUserAsync>d__19.MoveNext()

### Relevant code snippets

```csharp
services.AddAuthentication(options =>
{
	options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
	options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
})
.AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, options =>
{
	options.Audience = configuration["B2C:Audience"];
	options.Authority = configuration["B2C:Authority"];
	options.TokenValidationParameters = new TokenValidationParameters
	{
		NameClaimType = ClaimTypes.Email,
		RoleClaimType = ClaimTypes.Roles
	};
})
.AddMicrosoftIdentityWebApp(options =>
{
	configuration.Bind(B2COptions.Key, options);
	options.TokenValidationParameters.NameClaimTypeRetriever = (_, _) => ClaimTypes.Email;
	options.TokenValidationParameters.RoleClaimType = ClaimTypes.Roles;
	options.Events ??= new OpenIdConnectEvents();
	options.ClientSecret = configuration["B2C:ClientSecret"];
	options.Events.OnRedirectToIdentityProvider = OnRedirectToIdentityProvider;
}, optionsCookie =>
{
	optionsCookie.Cookie.Name = configuration["CookieName"];
	optionsCookie.Cookie.SameSite = Microsoft.AspNetCore.Http.SameSiteMode.Lax;
	optionsCookie.Cookie.IsEssential = true;
	optionsCookie.Events.OnRedirectToAccessDenied = context =>
	{
		context.Response.StatusCode = StatusCodes.Status403Forbidden;
		return Task.CompletedTask;
	};
})
.EnableTokenAcquisitionToCallDownstreamApi()
.AddDistributedTokenCaches();
```

### Regression

since `2.17.2` this started to happen. We're therefore staying on `2.17.1`.

### Expected behavior

calling the api being in the same tenant with the correct permissions and allowances of the scopes provided should return the bearer token and not result into an error

**Comment by EelcoLos** [comment]
tested today: 2.18.0 has this issue too

**Comment by jmprieur** [comment]
@EelcoLos do you mind sharing your appsettings.json? especially the authority?

**Comment by EelcoLos** [comment]
I found the issue I was having:
In the "B2C" options key, I registered both the instance/domain *AND* Authority. 

because both `services.AddAuthentication().AddJwtBearer()` as `AddMicrosoftIdentityWebApp()` uses the `B2C` Options key, this triggered the `generic` option here:
https://github.com/AzureAD/microsoft-identity-web/blob/fc5d24a4f54a57e2e58762e760474fc3effc659e/src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs#L653

we needed the authority for the `AddJwtBearer()`, so i moved that out of the `B2C` key and the issue is resolved.

**Comment by seaglass** [comment]
I ran into the same issue. Thank you @EelcoLos for your find there...

if anyone else is curious here is a little more background. It seems this change comes from adding support for custom user domains in this PR

https://github.com/AzureAD/microsoft-identity-web/pull/2696

So I guess the theory now is that if you are setting Authority in your ADB2C configuration settings you WANT it to be a custom domain and it falls back to GenericAuthority instead of B2CAuthority.

It seems like it was automatically building the URL anyway before with this code


```
if (mergedOptions.IsB2C)
 {
     authority = $"{mergedOptions.Instance}{ClaimConstants.Tfp}/{mergedOptions.Domain}/{mergedOptions.DefaultUserFlow}";
     builder.WithB2CAuthority(authority);
}
```

 so even if you were setting Authority it appears that it wasn't actually used for this. It's just that before the aforementioned PR it was benign.

TL;DR: Remove authority from your ADB2C config options unless you truly want this to be a custom domain.
